bindings/java/c/evmc-vm.c

```diff
@@ -0,0 +1,116 @@
+#include <stdlib.h>
+#include <assert.h>
+#include <stdint.h>
+
+#include "evmc/loader.h"
+#include "evmc/helpers.h"
+#include "evmc-vm.h"
+#include "host.h"
+
+JNIEXPORT jobject JNICALL Java_org_ethereum_evmc_EvmcVm_init(JNIEnv* jenv, jclass jcls, jstring jfilename)
+{
+    struct evmc_vm *evm;
+    jint rs = (*jenv)->GetJavaVM(jenv, &jvm);
+    assert (rs == JNI_OK);
+    // load the EVM
+    const char *filename = (*jenv)->GetStringUTFChars(jenv, jfilename, 0);
+    if(filename != NULL){
+        enum evmc_loader_error_code loader_error;
+        evm = evmc_load_and_create(filename, &loader_error);
+        if (evm == NULL || loader_error != EVMC_LOADER_SUCCESS) {
+            const char *error_msg = evmc_last_error_msg();
+            jclass jclazz = (*jenv)->FindClass(jenv, "java/lang/AssertionError");
+            (*jenv)->ThrowNew(jenv, jclazz, error_msg);
+        }
+        (*jenv)->ReleaseStringUTFChars(jenv, jfilename, filename);
+    } else {
+        jclass jclazz = (*jenv)->FindClass(jenv, "java/lang/AssertionError");
+        (*jenv)->ThrowNew(jenv, jclazz, "JNI Error: filename cannot be NULL. \n");
+    }
+     return (*jenv)->NewDirectByteBuffer(jenv, (void*) evm, sizeof(struct evmc_vm));
+}
+
+JNIEXPORT jint JNICALL Java_org_ethereum_evmc_EvmcVm_abi_1version(JNIEnv *jenv, jclass jcls){
+    return EVMC_ABI_VERSION;
+}
+
+JNIEXPORT jstring JNICALL Java_org_ethereum_evmc_EvmcVm_name(JNIEnv *jenv, jclass jcls, jobject jevm){
+    struct evmc_vm *evm = (struct evmc_vm *) (*jenv)->GetDirectBufferAddress(jenv, jevm);
+    const char *evm_name = evmc_vm_name(evm);
+    return (*jenv)->NewStringUTF(jenv, evm_name);
+}
+
+JNIEXPORT jstring JNICALL Java_org_ethereum_evmc_EvmcVm_version(JNIEnv *jenv, jclass jcls, jobject jevm){
+    struct evmc_vm *evm = (struct evmc_vm *) (*jenv)->GetDirectBufferAddress(jenv, jevm);
+    const char *evm_version = evmc_vm_version(evm);
+    return (*jenv)->NewStringUTF(jenv, evm_version);
+}
+
+JNIEXPORT void JNICALL Java_org_ethereum_evmc_EvmcVm_destroy(JNIEnv *jenv, jclass jcls, jobject jevm){
+    struct evmc_vm *evm = (struct evmc_vm *) (*jenv)->GetDirectBufferAddress(jenv, jevm);
+    evmc_destroy(evm);
+}
+
+JNIEXPORT jobject JNICALL Java_org_ethereum_evmc_EvmcVm_execute(JNIEnv *jenv, jclass jcls, jobject jevm, jint jcontext_index, jint jrev, jobject jmsg, jobject jcode, jint jsize){
+    jobject jresult;
+    struct evmc_result* result = malloc (sizeof (struct evmc_result));
+    struct evmc_message *cmsg = (struct evmc_message *) (*jenv)->GetDirectBufferAddress(jenv, jmsg);
```

Comment:
I think the best approach would be to add an `assert` for all these functions which can fail to acquire.